docs/workflow_tasks/discover_granules.md

```diff
@@ -0,0 +1,55 @@
+---
+id: discover_granules
+title: Discover Granules
+hide_title: true
+---
+
+# Discover Granules
+
+This task utilizes the Cumulus Message Adapter to interpret and construct incoming and outgoing messages.
+
+Links to the npm package, task input, output and configuration schema definitions, and more can be found on the auto-generated [Cumulus Tasks](../tasks) page.
+
+## Summary
+
+The purpose of this task is to facilitate ingest of data that does not conform to either a PDR/[SIPS](../data-cookbooks/sips-workflow) discovery mechanism, a [CNM Workflow](../data-cookbooks/cnm-workflow) or direct injection of workflow triggering events into Cumulus core components.
+
+The task utilizes a defined [collection](../data-cookbooks/setup#collections) in concert with a defined [provider](../data-cookbooks/setup#providers) to scan a location for files matching the defined collection configuration, assemble those files into groupings by granule, and passes the constructed granules object as an output.
+
+The constructed granules object is defined by the collection passed in the configuration, and has impacts to other provided core [Cumulus Tasks](../tasks).
+
+Users of this task in a workflow are encouraged to carefully consider their configuration  in context of downstream tasks and workflows.
+
+## Task Inputs
+
+Each of the following sections are a high-level discussion of the intent of the various input/output/config values.
+
+For the most recent config.json schema, please see the [Cumulus Tasks page](../tasks) entry for the schema.
+
+### Input
+
+This task does not expect an incoming payload.
```

Comment:
This may be not worth mentioning, but I think this could be misconstrued as the task expecting _no_ input, which is incorrect. Would it be worthwhile to make the distinction that it doesn't expect any user input, but _does_ expect a Cumulus Message as input?

Reply:
This is definitely implicit because of the configuration discussed after this entry. Completely up to you.

Reply:
Part of the problem is we have task input which contains an Input and configuration objects.    I could call it "Incoming Event" as a header instead of Input(s), but that might be misconstrued by users that don't really understand CMA. 

I really don't like either option 🙁